### PR TITLE
Panache

### PR DESCRIPTION
**Pull request by FroMage**
Now supports Panache entities.

Still not type-safe, but same as the current version.
Also missing support for panache repositories, but that's much harder if not type-safe.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -73,12 +81,14 @@ shadowJar {
...
     relocate 'groovyjarjarcommonscli', 'org.hibernate.query.validator.cli'
     relocate 'groovyjarjarpicocli', 'org.hibernate.query.validator.picocli'
+    classifier = 'fatjar'
 }
 
 publishing {
     publications {
         maven(MavenPublication) {
             from(components.java)
+            artifact shadowJar
```
Comment (gavinking) [inline build.gradle:91]: @FroMage nice, thanks for that!

File: src/main/java/org/hibernate/query/validator/EclipseProcessor.groovy [hunk 2]
```diff
@@ -276,6 +309,110 @@ class EclipseProcessor extends AbstractProcessor {
...
+            ErrorReporter handler = new ErrorReporter(stringLiteral, unit, compiler);
+            collectPanacheArguments(args);
+            int[] offset = new int[1];
+            String hql = PanacheUtils.panacheQlToHql(handler, targetType, methodName, 
+                                                     panacheQl, offset, setParameterLabels);
+            if(hql == null)
+                return;
+            if(!setOrderBy.isEmpty()) {
+                hql += " ORDER BY "+String.join(", ", setOrderBy);
+            }
```
Comment (gavinking) [inline src/main/java/org/hibernate/query/validator/EclipseProcessor.groovy:322]: @FroMage couldn't this order by clause stuff be sucked into `PanacheUtils`?

Comment (FroMage) [reply src/main/java/org/hibernate/query/validator/EclipseProcessor.groovy:322]: Good point.

Comment (FroMage) [reply src/main/java/org/hibernate/query/validator/EclipseProcessor.groovy]: Done.

**Comment by FroMage** [comment]
So, this now supports repositories and will only check inside panache entities or repositories.

I think it's a fair start.

Remaining issues:

- No support for named query checking:
```java
// this should be validatable, we should be able to know it's missing params
// but the ORM version doesn't do that either, and I wouldn't know where to place
// the error marker
find("#namedQuery")
```
- No support for queries outside the entity/repo (requires support from javac)
- Some error markers are wrong for Panache: `find("name > 0 > 0" will extend to the next line, it must be due to the errorOffset being wrong (since I artificially insert "from MyEntity where " I shift errors by the errorOffset) but I can't find how it finds the end of the error in Eclipse.

I think it's fine for a first version.

**Comment by FroMage** [comment]
I updated the README to mention Panache too.

**Comment by gavinking** [comment]
Merged, thanks @FroMage!

cc @emmanuelbernard Query Validator supports Panache now, so you have no excuse not to put it in Quarkus.

**Comment by emmanuelbernard** [comment]
Neat. Is there an issue @FroMage on the Quarkus side to follow up with the annotation process config changes? Otherwise I'll open one.

**Pull request merged**